src/hooks/useContributions.ts

```diff
@@ -30,53 +41,93 @@ export interface UserContributions {
 // FIXME ideally we'd show this in a Storybook story
 const mockData: UserContributions = {
   ['0x23f24381cf8518c4fafdaeeac5c0f7c92b7ae678']: [
-    { title: 'I did a thing', link: 'http://thing.com' },
-    { title: 'And then another', link: 'http://another.com' },
-    { title: 'And YET ANOTHER! O_O', link: 'http://yetanother.com' },
-    { title: 'Gib me tokens', link: 'http://gib.com' },
-    { title: 'I crushed it', link: 'http://crush.com' },
+    { title: 'I did a thing', link: 'http://thing.com', source: 'thing' },
+    {
+      title: 'And then another',
+      link: 'http://another.com',
+      source: 'another',
+    },
+    {
+      title: 'And YET ANOTHER! O_O',
+      link: 'http://yetanother.com',
+      source: 'yetanother',
+    },
+    { title: 'Gib me tokens', link: 'http://gib.com', source: 'gib' },
+    { title: 'I crushed it', link: 'http://crush.com', source: 'crush' },
   ],
 };
 
-export function useContributionUsers(): UserContributions {
+const ensureSource =
+  (source: string) =>
+  (res: Response): Response => {
+    return {
```

Comment:
you can return an object literal without using `return` if you wrap it in parentheses: `() => ({ foo: 'bar' })`

Reply:
with all the nesting this felt cleaner, but okay.